src/api-serverless/src/profiles/proxies/profile-proxies.routes.ts

```diff
@@ -0,0 +1,108 @@
+import { Request, Response } from 'express';
+import { asyncRouter } from '../../async.router';
+import {
+  DEFAULT_MAX_SIZE,
+  DEFAULT_PAGE_SIZE,
+  FullPageRequest,
+  Page,
+  PageSortDirection
+} from '../../page-request';
+import { ApiResponse } from '../../api-response';
+import { ProfileProxyEntity } from '../../../../entities/IProfileProxy';
+import * as Joi from 'joi';
+import { getValidatedByJoiOrThrow } from '../../validation';
+import { profileProxyApiService } from '../../proxies/proxy.api.service';
+import { profilesService } from '../../../../profiles/profiles.service';
+import { BadRequestException } from '../../../../exceptions';
+import {
+  getAuthenticatedWalletOrNull,
+  maybeAuthenticatedUser
+} from '../../auth/auth';
+
+const router = asyncRouter({ mergeParams: true });
+
+router.get(
+  '/received',
+  maybeAuthenticatedUser(),
+  async (
+    req: Request<
+      { handleOrWallet: string },
+      any,
+      any,
+      ProfileReceivedProfileProxiesQuery,
+      any
+    >,
+
+    res: Response<ApiResponse<Page<ProfileProxyEntity>>>
+  ) => {
+    // fix any
+    const maybeAuthenticatedWallet = getAuthenticatedWalletOrNull(req as any);
+    const maybeAuthenticatedProfile =
+      maybeAuthenticatedWallet
+        ? await profilesService.getProfileAndConsolidationsByHandleOrEnsOrIdOrWalletAddress(
+            maybeAuthenticatedWallet
+          )
+        : null;
+
+    const targetProfile =
+      await profilesService.getProfileAndConsolidationsByHandleOrEnsOrIdOrWalletAddress(
+        req.params.handleOrWallet
+      );
+    if (!targetProfile?.profile) {
+      throw new BadRequestException(
+        `Profile with id ${req.params.handleOrWallet} does not exist`
+      );
+    }
+
+    const isRequesterTarget =
+      maybeAuthenticatedProfile?.profile?.external_id ===
+      targetProfile.profile?.external_id;
+
+    console.log('isRequesterTarget', isRequesterTarget);
+
+    const unvalidatedQuery = req.query;
+    const query = getValidatedByJoiOrThrow(
+      unvalidatedQuery,
+      ProfileReceivedProfileProxiesQuerySchema
+    );
+
+    // make new interface what returns also actions and ProfileMin's, if target is not requester, return only active actions, return only (both cases) where there is at least 1 action after filter
+    const result =
+      await profileProxyApiService.getProfileReceivedProfileProxies({
+        target_id: targetProfile.profile.external_id,
+        ...query
+      });
+    res.send(result);
+  }
+);
+
+export interface ProfileReceivedProfileProxiesQuery
```

Comment:
IMO you should put this into profile-proxies.db.service.ts and include target_id as a field. In validator you can use Omit. And as profile-proxies.db.service.ts is a fully API bridge to DB, it should be moved from root to api-serverless/src/proxies